File: IService.java
package me.service;

import me.exceptii.ValidationException;
import me.model.*;

import java.util.ArrayList;
import java.util.List;

public interface IService {
     void login(Angajat angajat, IProbaObserver client) throws ServiceException;
     void logout(Angajat angajat, IProbaObserver client) throws ServiceException;
     void check(Angajat angajat) throws ServiceException;
     List<ProbaDTO> getProbeDTO() throws ServiceException;
     List<ParticipantDTO> getParticipantiDTOcuProbaP(int p) throws ServiceException;
     List<Participant> findParticipantiDupaNume(String nume) throws ServiceException;
     Angajat getAngajat(String username, String parola) throws ServiceException;
     void saveProbaParticipant(ArrayList<Integer> probe, Integer idParticipant) throws ServiceException, ValidationException;
     Participant saveParticipant(Integer id, String nume, int varsta) throws ServiceException, ValidationException;
}

File: Concurs Inot C#/IService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Problema_3_CSharp.me.model;

namespace Problema_3_CSharp.me.service
{
    public interface IService
    {
        Angajat SaveAngajat(int id, String nume, int varsta, String username, String parola);

        Angajat GetAngajatUsernameParola(String username, String parola);

        Proba SaveProba(int id, float distanta, String stil);

        Proba GetProba(int id);

        Participant SaveParticipant(int id, String nume, int varsta);

        Participant GetParticipant(int id);

        IEnumerable<Participant> GetParticipantiDupaNume(String nume);

        void SaveProbaParticipant(List<int> probe, int idParticipant, int probaCurenta);

        IEnumerable<ProbaParticipant> FindProbaParticipantDupaParticipant(int idParticipant);

        IEnumerable<ProbaParticipant> FindProbaParticipantDupaProba(int idProba);

        IEnumerable<ProbaDTO> GetProbeDTO();

        IEnumerable<ParticipantDTO> getParticipantiDTOCuProbaP(int p);

        void Login(Angajat a, IObserver client);

        void Logout(Angajat a, IObserver client);

        void Check(Angajat a);
    }
}
